rust/processor/src/models/mod.rs

```diff
@@ -16,3 +20,111 @@ pub mod token_models;
 pub mod token_v2_models;
 pub mod transaction_metadata_model;
 pub mod user_transactions_models;
+
+lazy_static! {
+    pub static ref EVENTS_MIGRATION_MAP: HashMap<&'static str, &'static str> = {
+        let mut m = HashMap::new();
+        m.insert(
+            "0x1::aptos_governance::VoteEvent",
+            "0x1::aptos_governance::Vote",
+        );
+        m.insert(
+            "0x1::stake::DistributeRewardsEvent",
+            "0x1::stake::DistributeRewards",
+        );
+        m.insert(
+            "0x1::delegation_pool::AddStakeEvent",
+            "0x1::delegation_pool::AddStake",
+        );
+        m.insert(
+            "0x1::delegation_pool::UnlockStakeEvent",
+            "0x1::delegation_pool::UnlockStake",
+        );
+        m.insert(
+            "0x1::delegation_pool::WithdrawStakeEvent",
+            "0x1::delegation_pool::WithdrawStake",
+        );
+        m.insert(
+            "0x1::delegation_pool::ReactivateStakeEvent",
+            "0x1::delegation_pool::ReactivateStake",
+        );
+        m.insert("0x3::token::MintTokenEvent", "0x3::token::MintToken");
+        m.insert("0x3::token::BurnTokenEvent", "0x3::token::BurnToken");
+        m.insert(
+            "0x3::token::MutateTokenPropertyMapEvent",
+            "0x3::token::MutateTokenPropertyMap",
+        );
+        m.insert("0x3::token::WithdrawEvent", "0x3::token::Withdraw");
+        m.insert("0x3::token::DepositEvent", "0x3::token::Deposit");
+        m.insert(
+            "0x3::token_transfers::TokenOfferEvent",
+            "0x3::token_transfers::TokenOffer",
+        );
+        m.insert(
+            "0x3::token_transfers::TokenCancelOfferEvent",
+            "0x3::token_transfers::TokenCancelOffer",
+        );
+        m.insert(
+            "0x3::token_transfers::TokenClaimEvent",
+            "0x3::token_transfers::TokenClaim",
+        );
+        m.insert("0x4::token::MutationEvent", "0x4::token::Mutation");
+        m.insert("0x1::object::TransferEvent", "0x1::object::TransferEvent");
+        m
+    };
+    pub static ref EVENTS_MIGRATION_PREFIX_MAP: HashMap<&'static str, &'static str> = {
+        let mut m = HashMap::new();
+        m.insert("0x1::coin::WithdrawEvent", "0x1::coin::Withdraw<");
```

Comment:
What's the rationale for having coin type generic in the event name? This makes it so much harder to index bc we need to parse out the generic part of the event name. Why not have coin_type as part of the event struct?